deployment/.gitignore

```diff
@@ -7,3 +7,5 @@ node_modules
 # CDK asset staging directory
 .cdk.staging
 cdk.out
+
+package-lock.json
```

Comment:
I wonder if this affect the overall deployment process when deploying this stack on aws

Reply:
The existing code base does not have this lock file https://github.com/opensearch-project/opensearch-build/tree/main/deployment

Also originally I had it in pr but it will fail the PR check. https://github.com/opensearch-project/opensearch-build/pull/1660#issuecomment-1046544618

Reply:
My bad, I was looking at package.json.
Thanks!